obspy/io/seiscomp/tests/test_event.py

```diff
@@ -34,69 +38,192 @@ def setUp(self):
-        catalog = _read_quakeml(filename)
+        sc3ml_doc = transform(etree.parse(filename))
 
         with NamedTemporaryFile() as tf:
-            tmpfile = tf.name
-            catalog.write(tmpfile, format='SC3ML', validate=validate,
-                          verbose=True, event_removal=event_removal)
+            tf.write(sc3ml_doc)
+            if validate:
+                self.assertTrue(_validate_sc3ml(tf.name))
             filepath_cmp = os.path.join(self.path, sc3ml_file)
-            self.assertTrue(filecmp.cmp(filepath_cmp, tmpfile))
+            self.assertTrue(filecmp.cmp(filepath_cmp, tf.name))
+
+    def test_read_xslt_event(self):
+        self.cmp_read_xslt_file('quakeml_1.2_event.sc3ml',
+                                'quakeml_1.2_event_res.xml')
+
+    def test_read_xslt_origin(self):
+        self.cmp_read_xslt_file('quakeml_1.2_origin.sc3ml',
+                                'quakeml_1.2_origin_res.xml')
+
+    def test_read_xslt_magnitude(self):
+        self.cmp_read_xslt_file('quakeml_1.2_magnitude.sc3ml',
+                                'quakeml_1.2_magnitude.xml')
+
+    def test_read_xslt_station_magnitude_contribution(self):
+        self.cmp_read_xslt_file(
+            'quakeml_1.2_stationmagnitudecontributions.sc3ml',
+            'quakeml_1.2_stationmagnitudecontributions.xml')
+
+    def test_read_xslt_station_magnitude(self):
+        self.cmp_read_xslt_file('quakeml_1.2_stationmagnitude.sc3ml',
+                                'quakeml_1.2_stationmagnitude.xml')
+
+    def test_read_xslt_data_used_in_moment_tensor(self):
+        self.cmp_read_xslt_file('quakeml_1.2_data_used.sc3ml',
+                                'quakeml_1.2_data_used.xml')
+
+    def test_read_xslt_arrival(self):
+        self.cmp_read_xslt_file('quakeml_1.2_arrival.sc3ml',
+                                'quakeml_1.2_arrival_res.xml')
+
+    def test_read_xslt_pick(self):
+        self.cmp_read_xslt_file('quakeml_1.2_pick.sc3ml',
+                                'quakeml_1.2_pick.xml')
+
+    def test_read_xslt_focalmechanism(self):
+        self.cmp_read_xslt_file('quakeml_1.2_focalmechanism.sc3ml',
+                                'quakeml_1.2_focalmechanism_res.xml')
+
+    def test_read_xslt_iris_events(self):
+        # Magnitude lost during conversion
+        self.cmp_read_xslt_file('iris_events.sc3ml', 'iris_events_res.xml')
+
+    def test_read_xslt_example(self):
+        self.cmp_read_xslt_file('qml-example-1.2-RC3.sc3ml',
+                                'qml-example-1.2-RC3.xml')
+
+    def test_read_example(self):
+        filename = os.path.join(self.path, 'qml-example-1.2-RC3.sc3ml')
+        catalog = _read_sc3ml(filename)
+
+        self.assertEqual(len(catalog.events), 1)
+        self.assertEqual(len(catalog.events[0].origins), 1)
 
-    def test_event(self):
-        self.cmp_file('quakeml_1.2_event.xml', 'quakeml_1.2_event.sc3ml')
+    def test_read_ID_prefix(self):
```

Comment:
Guess you'll have to change it to `test_read_id_prefix` to make flake8 happy (or add `  # NOQA` but renaming is not a problem so better not circumvent flake8)